src/app/components/layouts/portlet/terra-portlet.component.spec.ts

```diff
@@ -1,11 +1,265 @@
+
+        debugElement.query(By.css('div.portlet-head')).triggerEventHandler('click', null);
+        fixture.detectChanges();
+
+        expect(debugElement.query(By.css('div.portlet-body')).classes['collapsed']).toBe(true);
+
+        debugElement.query(By.css('div.portlet-head')).triggerEventHandler('click', null);
+        fixture.detectChanges();
+
+        expect(debugElement.query(By.css('div.portlet-body')).classes['collapsed']).toBe(false);
+    });
+
+    it(`should update view correctly if 'inputIsCollapsable' is changed`, () =>
+    {
+        component.inputPortletHeader = portletHeader;
+        component.inputIsCollapsable = true;
+        fixture.detectChanges();
+
+        component.inputIsCollapsable = false;
+        component.inputCollapsedChange.subscribe((value:boolean) => expect(value).toBe(false));
+
+        component.ngOnChanges({inputIsCollapsable: new SimpleChange(true, false, true)});
+
+        expect(component.inputCollapsed).toBe(false);
+    });
+
+    it(`[NEED BETTER EXPECTATION] should toggles 'inputCollapsed' and emits 'inputCollapsedChange' if 'inputIsCollapsable is true and #toggleCollapse() is called`,
+        () =>
+        {
+            let inputCollapsedChangeEmit:Spy = spyOn(component.inputCollapsedChange, 'emit');
+
+            component.inputPortletHeader = portletHeader;
+            component.inputIsCollapsable = true;
+            component.inputCollapsed = false;
+            fixture.detectChanges();
+
+            expect(component.inputCollapsed).toBe(false);
+
+            component.toggleCollapse();
+
+            expect(component.inputCollapsed).toBe(true);
+            expect(inputCollapsedChangeEmit).toHaveBeenCalledWith(true);
+
+            component.toggleCollapse();
+
+            expect(component.inputCollapsed).toBe(false);
+            expect(inputCollapsedChangeEmit).toHaveBeenCalledWith(false);
+        });
+
+    it(`should set'inputCollapsed' to false when #toggleCollapse() is called and 'inputIsCollapsable is false`, () =>
+    {
+        component.inputPortletHeader = portletHeader;
+        component.inputIsCollapsable = true;
+        component.inputCollapsed = true;
+        fixture.detectChanges();
+
+        expect(component.inputCollapsed).toBe(true);
+
+        component.inputIsCollapsable = false;
+
+        component.toggleCollapse();
+
+        expect(component.inputCollapsed).toBe(false);
+    });
+
+    it(`should render a given 'inputButtonList' correctly`, () =>
+    {
+        component.inputPortletHeader = portletHeader;
+        component.inputButtonList = buttonList;
+        fixture.detectChanges();
+
+        let htmlButtonDebugElements:Array<DebugElement> = debugElement.queryAll(By.css('div.header-buttons terra-button'));
+
+        expect(htmlButtonDebugElements.length).toBe(buttonList.length);
+
+        htmlButtonDebugElements.forEach((button:DebugElement, index:number) =>
```

Comment:
vielleicht könnte man die Funktionsdeklaration noch in ein `beforeAll()` packen um sie im Setup ersichtlich zu haben

Reply:
macht doch nicht wirklich Sinn -> closed